dynatrace_event/stackstate_checks/dynatrace_event/dynatrace_event.py

```diff
@@ -0,0 +1,264 @@
+# (C) StackState 2020
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from stackstate_checks.base import AgentCheck, ConfigurationError, TopologyInstance
+from .util import DynatraceData
+
+import requests
+from requests import Session
+import yaml
+from datetime import datetime, timedelta
+import time
+
+
+class DynatraceEventCheck(AgentCheck):
+
+    INSTANCE_TYPE = "dynatrace_event"
+    SERVICE_CHECK_NAME = "dynatrace_event"
+
+    def __init__(self, name, init_config, instances=None):
+        AgentCheck.__init__(self, name, init_config, instances)
+        self.url = None
+        self.token = None
+        self.tags = None
+        self.environment = None
+        self.events_boostrap_days = None
+        self.events_process_limit = None
+        self.domain = None
+        self.status = None
+        self.verify = None
+        self.cert = None
+        self.keyfile = None
+        self.reset_run_state = False
+
+    def load_status(self):
+        """
+        Load the status from memory on each run if any events processed and stored
+        otherwise status will be empty
+        """
+        self.status = DynatraceData.load_latest_status()
+        if self.status is None:
+            self.status = DynatraceData()
+
+    def get_instance_key(self, instance):
+        if 'url' not in instance:
+            raise ConfigurationError('Missing API url in configuration.')
+
+        return TopologyInstance(self.INSTANCE_TYPE, instance["url"])
+
+    def check(self, instance):
+        """
+        Integration logic
+        """
+        if 'url' not in instance:
+            raise ConfigurationError('Missing API user in configuration.')
+        if 'token' not in instance:
+            raise ConfigurationError('Missing API Token in configuration.')
+
+        self.url = instance.get('url')
+        self.token = instance.get('token')
+        self.events_boostrap_days = instance.get('events_boostrap_days', 5)
+        self.events_process_limit = instance.get('events_process_limit', 10000)
+        self.tags = instance.get('tags', [])
+        self.environment = instance.get('environment', 'production')
+        self.verify = instance.get('verify', True)
+        self.cert = instance.get('cert', '')
+        self.keyfile = instance.get('keyfile', '')
+        self.load_status()
+        self.log.info("After loading the status: {}".format(self.status.data))
+        self.log.info("After loading, the from time is {}".format(self.status.data.get(self.url)))
+
+        try:
+            self.process_events()
+        except Exception as e:
+            self.log.exception(str(e))
+            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, tags=self.tags, message=str(e))
```

Comment:
shouldn't we also put it back to `AgentCheck.CLEAR` at some point?

Reply:
done.